src/presign/participant.rs

```diff
@@ -198,6 +229,10 @@ impl Input {
             .filter(|item| *item.participant() != pid)
             .collect()
     }
+    /// Returns the [`AuxInfoPublic`]s associated with all the participants
+    fn all_auxinfo_public(&self) -> Vec<AuxInfoPublic> {
+        self.all_auxinfo_public.clone()
```

Comment:
If you want this to return a `Vec` I don't think you can, because `self` owns the `Vec` we're holding. Even if you were okay with not-a-vec (like, you could get an array `&[AuxInfoPublic]` by calling `self.all_auxinfo_public.as_slice()`) I think once you get the data, you have to sort it, which means we're gonna move the bytes around and therefore need our own copy.

Reply:
You might update the docs to reflect that a bit more explicitly, like

> Returns a copy of the [`AuxInfoPublic`]s...